src/mds/Server.cc

```diff
@@ -1410,6 +1420,9 @@ void Server::reply_client_request(MDRequestRef& mdr, MClientReply *reply)
     mds->logger->inc(l_mds_reply);
     utime_t lat = ceph_clock_now() - mdr->client_request->get_recv_stamp();
     mds->logger->tinc(l_mds_reply_latency, lat);
+    if (g_conf->mds_ops_perf_counter) {
+      mds_op_count(req->head.op, lat);
```

Comment:
I am concerned that this early_reply+reply_client_request are not going to cover all the same paths as the hook in respond_to_request covered.

Especially, it looks like lots of the error handling cases will be ignored this way.

Is it intentional?  I think it is probably useful to have the operation counters increment for handling every request of a particular op type, even if we're going through a respond_to_request with an error rather than a full completion.

Reply:
Yes, i statistics all operations latency according to the l_mds_reply_latency.I want to get each operation as a percentage of total operating time easily.